http/transport/circuit_breaker_tripper.go

```diff
@@ -0,0 +1,89 @@
+package transport
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sony/gobreaker"
+	"golang.org/x/xerrors"
+)
+
+// circuitBreakerTripper implements a ChainableRoundTripper with
+// the circuit breaker "gobreaker". By default the circuit breaker
+// opens if more than 5 consecutive calls to RoundTrip fail while
+// it is closed. To change theses settings a gobreaker.Settings
+// object can be passed to NewCircuitBreakerTripper(). For more
+// information about the specifiable settings please visit:
+// https://github.com/sony/gobreaker
+//
+// To keep track how often the circuit breaker tripped and
+// transitioned to the open/half-open state, a prometheus counter
+// is added to each newly instantiated circuit breaker.
+//
+// CAUTION: It is advised to use this RoundTripper as the last used
+// RoundTripper before the actual request is triggered. Otherwise
+// the circuit breaker might not open because of real connectivity
+// problems, but because of other RoundTrippers.
+type circuitBreakerTripper struct {
+	transport http.RoundTripper
+	breaker   *gobreaker.CircuitBreaker
+}
+
+func NewDefaultCircuitBreakerTripper(name string) *circuitBreakerTripper {
+	return NewCircuitBreakerTripper(gobreaker.Settings{
+		Name: name,
+	})
+}
+
+func NewCircuitBreakerTripper(settings gobreaker.Settings) *circuitBreakerTripper {
+	if settings.Name == "" {
+		panic("name is mandatory for circuit breaker")
+	}
+
+	stateSwitchCounterVec := prometheus.NewCounterVec(prometheus.CounterOpts{
+		ConstLabels: prometheus.Labels{"name": settings.Name},
```

Comment:
What happens if you register the second circuit breaker? Do we then have both or does it error because it exists already?

Reply:
If the `name` is the same, the second circuit breaker will write to the same `CounterVec`. See https://github.com/pace/bricks/blob/cdade781f7055b617e75c45f3f608d7a9fc47cd1/http/transport/circuit_breaker_tripper.go#L53